Documentation/EventHub/BuildingExtensions.md

```diff
@@ -299,6 +314,21 @@ All extensions are provided a default API to read XDM shared state from another
 func getXDMSharedState(extensionName: String, event: Event?) -> SharedStateResult?
 ```
 
+If you want more control over the reading of the shared state, you can use the overloaded version of the above method:
+
+```swift
+/// Gets the XDM SharedState data for a specified extension. If this extension populates multiple mixins in their shared state, all the data will be returned at once and it can be accessed using path discovery.
+/// - Parameters:
+///   - extensionName: An extension name whose `SharedState` will be returned
+///   - event: If not nil, will retrieve the `SharedState` that corresponds with the event's version, if nil will return the latest `SharedState`
+///   - barrier: If true, the `EventHub` will only return `.set` if `extensionName` has moved past `event`
+///   - resolution: The `SharedStateResolution` to resolve for
+/// - Returns: A `SharedStateResult?` for the requested `extensionName` and `event`
+func getXDMSharedState(extensionName: String, event: Event?, barrier: Bool = false, resolution: SharedStateResolution = .any) -> SharedStateResult?
+```
+
+The resolution is used to fetch a specific type of shared state. Using `.any` will just fetch the last shared state, but using `.lastSet`, will fetch the last shared state with a `.set` status. This is useful if you would like to read the cached config before the remote config has been downloaded. 
```

Comment:
Suggest rewording: "Using `.any` will fetch the last shared state with any status, while using `.lastSet` will fetch the last shared state with status of `.set`."